### ISSUE TITLE
Markdown in RawHTML Fails When &thinsp; Is Present

### ISSUE THREAD

**Issue by tajmone** [jgm/pandoc#4088]
pandoc v2.0.2

I use a custom PP macro to emulate GFM Task-Lists in pandoc markdown:

https://github.com/tajmone/pandoc-goodies/tree/master/pp#gfm-task-lists

Up to pandoc v2.0 it always worked fine, now an issue has arised: markdown is no longer interpreted as expected.

This PP macro:

```
!TaskList(
!Task[x][I'm a _checked_ task]
)
```

... emits the following Raw HTML in the source doc, before feeding it to pandoc:

``` html
<ul class="task-list">
<li class="task-list-item">
<input type="checkbox" disabled="" checked="">&thinsp;I'm a _checked_ task</li>
</ul>
```

Since pandoc v2 it looks like the emphasis in `_checked_ task` is not being interpreted as markdown. 

The problem seems to be the presence of the `&thinsp;` entity, because if I remove it from the macro all text gets formatted as expected. Even putting a normal space between  `&thinsp;` and the text doesn't solve the issue --- looks like pandoc's parser is set off by the thin-space entity.

**Comment by tajmone** [comment]
**NOTE** — As a temporary workaround, I've changed the `&thinsp;` in the macro with a  backslash-escaped space "`\ `". This works, but the desired character was a thin-space, not a nonbreaking space, so it's just a temporary solution.

I haven't testes which HTML entities break up markdown rendering in raw HTML, but I've tried using `&#8201;` instead, and failed too.

I'm not sure if the problem is that the source document goes first through the PP preprocessor, which pipes the output to pandoc via STDOUT; but with pandoc v1.x this issue didn't occur, so something must have changed in how pandoc handles this.
